src/bin/pg_autoctl/keeper.c

```diff
@@ -488,19 +509,33 @@ keeper_update_pg_state(Keeper *keeper)
 	 */
 	switch (keeperState->current_role)
 	{
-		case PRIMARY_STATE:
 		case WAIT_PRIMARY_STATE:
+		{
+			/* we don't expect to have a streaming replica */
+			return postgres->pgIsRunning;
+		}
+
+		case PRIMARY_STATE:
+		{
+			/* we expect replication to be in place */
+			return postgres->pgIsRunning
+				&& !IS_EMPTY_STRING_BUFFER(postgres->currentLSN)
```

Comment:
why we make emptiness check for lsn and syncstate is not explicit. perhaps single line explaining why

Reply:
Good point, done now!